### ISSUE TITLE
docker machine provisions `docker daemon` instead of `dockerd`

### ISSUE THREAD

**Issue by blaggacao** [docker/machine#4202]
#### docker-machine version
```
docker-machine --version
docker-machine version 0.12.0, build 45c69ad
```
#### Target machine
```
$ lsb_release -ra
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.2 LTS
Release:        16.04
Codename:       xenial
```
#### Bug observations
While `/lib/systemd/system/docker.service` contains this line
`ExecStart=/usr/bin/dockerd -H fd://`

`/etc/systemd/system/docker.service.d/10-machine.conf` shows:
```
ExecStart=/usr/bin/docker daemon -H tcp://0.0.0.0:2376 -H unix:///var/run/docker.sock --storage-driver aufs --tlsverify --tlscacert /etc/docker/ca.pem --tlscert /etc/docker/server.pem --tlskey /etc/docker/server-key.pem --label provider=generic
```
#### ! note `/usr/bin/docker daemon` !
which fails, as in fact `Docker version 17.06.0-ce, build 02c1d87` !< 1.12.0

#### First investigation
A quick look a the code tells me that probably something before this line is failing:
https://github.com/docker/machine/blob/master/libmachine/provision/systemd.go#L50

#### Further instructions required
What more information could be useful to track this down?

**Comment by dobsong** [comment]
I get the same thing with docker-machine version 0.12.2 as well

**Comment by dobsong** [comment]
Duplicate of https://github.com/docker/machine/issues/4156?

**Comment by bhuisgen** [comment]
Why this issue and #4156 are closed ? The bug is not fixed in 0.12.2.

'docker daemon' command has been deprecated since docker v17.05.0 so we need to replace by dockerd.
